### PR TITLE
openshift_adm_migrate_template_instances -  new module

### PR DESCRIPTION
**Pull request by alinabuzachis**
`openshift_adm_migrate_template_instances` -  new module to update TemplateInstances to point to the latest group-version-kinds.

### CODE REVIEW

**Review by Akasurde — CHANGES_REQUESTED** [review]
minor fixes. Rest LGTM

File: plugins/modules/openshift_adm_migrate_template_instances.py [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
```
[the remark this answers is not in the document]
Comment (alinabuzachis) [reply plugins/modules/openshift_adm_migrate_template_instances.py]: @Akasurde Thank you. Done!

File: plugins/modules/openshift_adm_migrate_template_instances.py [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+    argument_spec["namespace"] = dict(type="str")
+
+    return argument_spec
+
+
+def main():
+    argument_spec = argspec()
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
+    openshift_process = OpenShiftMigrateTemplateInstances(module)
+    openshift_process.argspec = argument_spec
```
Comment (abikouo) [inline plugins/modules/openshift_adm_migrate_template_instances.py:394]: I dont really see where this is used

**Review by abikouo — APPROVED** [review]
Good Job @alinabuzachis !!!

**Pull request merged**